libs/wingsdk/src/core/inflight.ts

```diff
@@ -53,3 +62,204 @@ export class InflightClient {
+ */
+export function inflight<TFunction extends AsyncFunction>(
+  fn: (ctx: {}, ...args: Parameters<TFunction>) => ReturnType<TFunction>
+) {
+  return new Lifter().inflight(fn);
+}
+
+/**
+ * Manages the liftable objects and operations for an inflight function.
+ */
+class Lifter<
+  TLifted extends Record<string, any>,
+  TOperations extends Record<string, string[]>
+> {
+  constructor(
+    private lifts: LiftableMap = {},
+    private grants: Record<string, string[]> = {}
+  ) {}
+
+  /**
+   * Add additional liftable objects to the scope of the inflight function.
+   * Any existing liftable objects with the same name will be overwritten.
+   *
+   * Conventionally, this is used by passing in a `const` object to bind it with the same name
+   *
+   * ```ts
+   * const bucket = new cloud.Bucket(app, "Bucket");
+   * const number = 5;
+   *
+   * lift({ bucket, number })
+   *   .inflight(({ bucket, number }) => { ... }))
+   * ```
+   *
+   * However, the name is not required to match the variable in the current scope.
+   *
+   * This is especially useful/necessary when lifting data via a reference or some other expression
+   *
+   * ```ts
+   * const bucket = new cloud.Bucket(app, "Bucket");
+   *
+   * lift({ bkt: bucket, sum: 2 + 2, field: bucket.field })
+   *   .inflight(({ bkt, sum, field }) => { ... }))
+   * ```
+   */
+  public lift<TWillLift extends LiftableMap>(captures: TWillLift) {
+    return new Lifter<
+      Omit<TLifted, keyof TWillLift> & LiftedMap<TWillLift>,
+      TOperations
+    >(
+      {
+        ...this.lifts,
+        ...captures,
+      },
+      this.grants
+    );
+  }
+
+  /**
+   * Grant permissions for lifted resources.
+   *
+   * By default, all all possible methods are granted to lifted resources.
+   * This function restricts those:
+   *
+   * ```ts
+   * const bucket = new cloud.Bucket(app, "Bucket");
+   *
+   * lift({ bucket })
+   *   .grant({ bucket: ["get"] })
+   *   .inflight(({ bucket }) => {
+   *     await bucket.get("key");
+   *     await bucket.set("key", "value"); // Error: set is not granted
+   *   });
+   * ```
+   *
+   * fields are always accessible, even if not granted.
+   */
+  public grant<
+    TNewOps extends Partial<{
+      [K in keyof TLifted]: OperationsOf<TLifted[K]>;
+    }>
+  >(grants: TNewOps) {
+    return new Lifter<TLifted, Omit<TOperations, keyof TNewOps> & TNewOps>(
+      this.lifts,
+      {
+        ...this.grants,
+        ...grants,
+      }
+    );
+  }
+
+  /**
+   * Create an inflight function with the available lifted data.
+   *
+   * This function must not reference any variables outside of its scope.
+   * If needed, use `lift` again to bind variables to the scope of the function.
+   * Bound variables will be available as properties on the `ctx` object passed as the first argument to the function.
+   *
+   * Built-in NodeJS globals are available, such as `console` and `process`.
+   */
+  public inflight<TFunction extends AsyncFunction>(
```

Comment:
Can we replace usage of `Testing.makeHandler()` with this and get rid of `makeHandler()`?